example/ConnectionInfoFetch.tsx

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+import * as React from 'react';
+import {Text, View, TouchableOpacity} from 'react-native';
+import NetInfo from '../src';
+
+interface State {
+  connectionInfo: string;
+}
+
+export default class ConnectionInfoCurrent extends React.Component<{}, State> {
+  state = {
+    connectionInfo: 'Tap to get current state',
+  };
+
+  componentDidMount() {
+    this._fetchState();
+  }
+
+  _fetchState = async () => {
+    const state = await NetInfo.fetch();
+    this.setState({
+      connectionInfo: JSON.stringify(state),
+    });
+  };
+
+  render() {
+    return (
+      <View>
+        <TouchableOpacity onPress={this._fetchState}>
+          <Text>{this.state.connectionInfo}</Text>
+        </TouchableOpacity>
+      </View>
+    );
+  }
+}
```

Comment:
Hey @matt-oakes
Question: why not writing those as function components with hooks instead? 
_something like this (haven't tested/ran, just converted class to func)_
```jsx
export default function ConnectionInfoCurrent() {
  const [state, setState] = React.useState('Tap to get current state');
  React.useEffect(() => {
    (async () => {
      await _fetchState();
    })();
  }, []);

  _fetchState = async () => {
    const state = await NetInfo.fetch();
    setState(JSON.stringify(state));
  };

  render() {
    return (
      <View>
        <TouchableOpacity onPress={_fetchState}>
          <Text>{state}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
```

Reply:
Most of these examples were written before React Hooks were available in React Native. We also have the `useNetInfo` hook itself so I want to encourage people using hooks to just use that and show anyone not wanting to use hooks how the standard API can be used.